cpp/src/datetime/datetime_ops.cu

```diff
@@ -357,6 +358,30 @@ std::unique_ptr<column> day_of_year(column_view const& column,
   return detail::apply_datetime_op<detail::extract_day_num_of_year, cudf::type_id::INT16>(
     column, stream, mr);
 }
+
+std::unique_ptr<column> is_leap_year(column_view const& column,
+                                     rmm::cuda_stream_view stream,
+                                     rmm::mr::device_memory_resource* mr)
+{
+  CUDF_EXPECTS(is_timestamp(column.type()), "Column type should be timestamp");
+
+  auto years      = detail::extract_year(column, stream);
+  auto years_view = years->view();
```

Comment:
There's no need to materialize the column of years. You can extract the year on the fly inside your transform kernel below.